src/uct/base/uct_md.c

```diff
@@ -384,17 +385,115 @@ ucs_status_t uct_rkey_release(uct_component_h component,
     return component->rkey_release(component, rkey_ob->rkey, rkey_ob->handle);
 }
 
+void uct_md_copy_v1_fields(uct_md_attr_t *dst, uct_md_attr_v2_t *src)
+{
+    dst->cap.max_alloc        = src->max_alloc;
+    dst->cap.max_reg          = src->max_reg;
+    dst->cap.flags            = src->flags;
+    dst->cap.reg_mem_types    = src->reg_mem_types;
+    dst->cap.detect_mem_types = src->detect_mem_types;
+    dst->cap.alloc_mem_types  = src->alloc_mem_types;
+    dst->cap.access_mem_types = src->access_mem_types;
+    dst->reg_cost             = src->reg_cost;
+    dst->rkey_packed_size     = src->rkey_packed_size;
```

Comment:
I think we should check `field_mask` or have `ucs_assert` that some field was filled by v2, e.g.
```
#define UCT_MD_ATTR_SET_V1_FIELD_FROM_V2(_md_attr_v1, _md_attr_v2, _field_name, _field_flag) \
do { \
    ucs_assert((_md_attr_v2)->field_mask & _field_flag); \
    (_md_attr_v1)->_field_name =  (_md_attr_v2)->_field_name; \
} while (0)
```